### PR TITLE
Add print pre-check to `handlebars`, `html`, `markdown`, `yaml`

### PR DESCRIPTION
**Pull request by fisker**
## Checklist

- [ ] I’ve added tests to confirm my change works.
- [ ] (If changing the API or CLI) I’ve documented the changes I’ve made (in the `docs/` directory).
- [ ] (If the change is user-facing) I’ve added my changes to `changelog_unreleased/*/XXXX.md` file following `changelog_unreleased/TEMPLATE.md`.
- [ ] I’ve read the [contributing guidelines](https://github.com/prettier/prettier/blob/main/CONTRIBUTING.md).

**✨[Try the playground for this PR](https://prettier.io/playground-redirect)✨**

### CODE REVIEW

File: src/language-markdown/printer-markdown.js [hunk 1]
```diff
@@ -456,7 +464,12 @@ function genericPrint(path, options, print) {
     case "export": // transformed in to `importExport`
     default:
       /* istanbul ignore next */
-      throw new Error(`Unknown markdown type ${JSON.stringify(node.type)}`);
+      throw Object.assign(
+        new Error(
+          "Unexpected markdown node type: " + JSON.stringify(node.type)
+        ),
+        { node }
+      );
```
Comment (thorn0) [inline src/language-markdown/printer-markdown.js:472]: How about extracting this to `UnexpectedNodeTypeError`?

Comment (fisker) [reply src/language-markdown/printer-markdown.js]: Used shorter name `UnexpectedNodeError` [`4098457` (#13392)](https://github.com/prettier/prettier/pull/13392/commits/4098457e654d36cc098bb7a225087cc77f2ab427)

Comment (thorn0) [reply src/language-markdown/printer-markdown.js]: How do we decide whether a file goes to `src/common` or to `src/utils`? Other errors live in `src/common/errors.js`.

Comment (fisker) [reply src/language-markdown/printer-markdown.js]: Mentioned once long time ago, https://github.com/prettier/prettier/pull/5889#pullrequestreview-206353103

My original plan is to move `src/common` into `src/utils`, but things didn't go well https://github.com/prettier/prettier/pull/13346

**Review by thorn0 — APPROVED** [review]

File: src/language-markdown/printer-markdown.js (hunk 1 above)
Comment (thorn0) [reply src/language-markdown/printer-markdown.js]: So you wanted to remove `common` completely and have `utils` instead? That would be just a rename, so I don't think that's what you mean. What were you going to keep in `common`?

Comment (fisker) [reply src/language-markdown/printer-markdown.js]: > So you wanted to remove common completely and have utils instead?

I mean extract all staff from `src/common/` into `src/utils/`, remove `src/common/` completely, and re-export in `src/utils/index.js`.

**Pull request merged**